### PR TITLE
support time range template variable

### PR DESCRIPTION
**Pull request by mtanda**
Implement https://github.com/grafana/grafana/issues/1909

### CODE REVIEW

**Comment by mrshaun13** [comment]
Would this request allow you to use a template selected variable in the time declaration in the query? Use case is to have a single value selected in a template in multiple parts of a query:

ie.
$dataSummary is selected in a template(values are 1,5,10,30,60,120) and specified to the end user that the values are in minutes
Query: SELECT sum("bytes")/$dataSummary*60 FROM "database" GROUP BY time($dataSummary m)

This allows an end user to group data into larger buckets than 1minutes and still display large amounts of historical data in a Bytes/Sec display.

**Comment by mtanda** [comment]
This PR provides unix epoch time of graph time range.
For example, from=1462287600000 (May 4 2016 00:00:00), to=1462309200000 (May 4 2016 06:00:00).

If you want to change data summary range (for InfluxDB?), you can use custom type template variable.
http://docs.grafana.org/reference/templating/

**Comment by mrshaun13** [comment]
The portion I **was** having difficulty with was being able concatenate a variable name next to a character to be able to use a single template variable to apply math + time dynamics to the query. 

This was not working: ($summary % 'm') 
This does work: ([[summary]]m)

**Comment by mtanda** [comment]
If you have trouble with concatenating template variable and other, you can use "[[var name]]" notation.
Your issue seems to be solved. 👍

**Comment by mtanda** [comment]
I think the use case of this feature is "making dashboard link with time range parameter in text panel".
The use case is very limited, and not required for many user.
So I close this PR.
(If someone want like this feature, please create datasource plugin and implement this feature as metricFindQuery.)

**Comment by cray74** [comment]
Actually, we do have another very important (at least for us) use case for this PR: we're recording IIoT data in InfluxDB, graphing it with Grafana; next to the actual measurments recorded in the database every 100ms, the SET-values are only written to the DB when they change, for the obvious efficiency reasons. Now, if you some into a time range, where the first SET-Value is only recorded somewhere in "the middle" of this time range, the chart line would be cut off, since the last valid SET-value is outside the time range.

If we had access to the $FROM (beginning of the the $timeFilter), we could simply add another query like this (pseudo code): SELECT LAST (value) from *.SET WHERE time <= $FROM.

This would yield the last valid value required to draw a continuous line, in a rather efficient way.

Summing up, we would really appreciate if you could update your PR to work with the most recent Grafana release. Mid-term, we really hope to get this feature in the mainstream release (noob question: how can we help for that to happen?).

**Comment by demospace** [comment]
This is an extremely useful feature! Was this PR rejected by the Grafana team?
